sklearn/tests/test_cross_validation.py

```diff
@@ -1038,3 +1039,57 @@ def test_check_is_partition():
 
     p[0] = 23
     assert_false(cval._check_is_partition(p, 100))
+
+
+def test_disjoint_label_folds():
+    ## Check that the function produces equilibrated folds
+    ##      with no label appearing in two different folds
+
+    # Fix the seed for reproducibility 
+    rng = np.random.RandomState(0)
+    
+    # Parameters of the test
+    n_labels = 15
+    n_samples = 1000
+    n_folds = 5
+    
+    # Construct the test data
+    tolerance = 0.05 * n_samples # 5 percent error allowed
```

Comment:
It seems a bit magic to me, it is not clear why the test should pass. It depends on the distribution of labels, right? I guess with randint and many samples, they will be quite close.